design-proposals/hardening-vmim.md

```diff
@@ -0,0 +1,125 @@
+# Overview
+The request to live migrate a VM is represented by an instance of VirtualMachineInstanceMigration kind.
+VirtualMachineInstanceMigration is a namespaced CRD and its instances are expected to be in the namespace of the VM they are referring to.
+As for today, by default, a namespace admin (usually a namespace "owner" in a less formal definition) is able to create VMs and also VirtualMachineInstanceMigrations (VMIMs) objects to enqueue a live migration request for a VM within his namespace.
+On the other side, live migration is also a building block for infrastructure critical operations like node drains or upgrades, so potentially every namespace admin allowed to create or delete VMIMs objects can delay or even prevent a cluster critical operation such as an upgrade.
+This proposal is about amending KubeVirt default RBAC roles according to the principle of least privilege so that namespace admins will be allowed to create or delete VirtualMachineInstanceMigrations objects only if explicitly allowed by a cluster admin.
+The present proposal is not an API change since the API will remain exactly the same, the change is only about who is allowed by default to call the API.
+
+## Motivation
+Live migration is an essential building block for cluster critical operations like node drains or upgrades.
+A less privileged user should not be allowed by default to be able to potentially interfere with infrastructure critical operations.
+
+## Goals
+- Namespace admins will not be allowed anymore by default to enqueue or delete migration request for the VMs in their namespaces
+- If needed, the right to enqueue or delete migration requests could be still granted by cluster admins to individual users or groups
+
+## Non Goals
```

Comment:
I would add that building a priority system is also out of scope as that would address some of the problems mentioned in this proposal.

Reply:
done